### PR TITLE
Tweak E2E test

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

**Comment by bergundy** [comment]
Integration eventually passed here: https://ci.binaris.io/blue/organizations/jenkins/gobinaris%2Fprecious/detail/feature-app-e2e/8/pipeline

File: app-testsuite/src/run.ts [hunk 1]
```diff
@@ -240,26 +240,31 @@ class App {
...
-          shell,
-          env: {
-            ...process.env,
-            CYPRESS_baseUrl: baseUrl,
-          },
-        });
+    const app = await createApp(testDir);
+    const runMode = process.env.APP_E2E_RUN_MODE || 'local';
+    const run = runMode === 'remote' ? app.runRemote : app.runLocal;
+    await run.bind(app)(async (baseUrl) => {
```
Comment (vogre) [inline app-testsuite/src/run.ts]: `runMode` can be "bound" into the app as a property.

File: app-testsuite/src/run.ts [hunk 2]
```diff
@@ -12,8 +12,8 @@ const shell = process.platform === 'win32';
 const ROOT_DIR = path.resolve(__dirname, '..', '..');
 const REGISTRY_URL = 'http://localhost:4873/';
 
-function log(...args: any[]) {
-  console.log(new Date(), ...args);
+function log(msg: string, ...args: any[]) {
+  console.log(`${new Date().toISOString()} ${msg}`, ...args);
```
Comment (vogre) [inline app-testsuite/src/run.ts]: Very appreciated.

File: app-testsuite/src/run.ts [hunk 3]
```diff
@@ -144,7 +144,7 @@ function publishToLocalRegistry(token: string) {
   });
 }
 
-async function withRegistry(testDir: string, fn: () => Promise<void>) {
+async function withRegistry(testDir: string, fn: () => Promise<any>) {
```
Comment (vogre) [inline app-testsuite/src/run.ts]: `withRegistry<T>` => `Promise<T>`?

Comment (bergundy) [reply app-testsuite/src/run.ts]: Good point

File: app-testsuite/src/run.ts (hunk 1 above)
Comment (bergundy) [reply app-testsuite/src/run.ts]: Changed a bit

**Pull request merged**